### PR TITLE
feat(transactions): bump Alloy + simplify `transfer_eth` example 

### PR DESCRIPTION
**Pull request by zerosnacks**
## Motivation

Bumps Alloy

Better show best practices and simplest API

Used for embedding in https://alloy.rs/highlights/the-transaction-lifecycle.html

## PR Checklist

- [ ] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

File: examples/transactions/examples/transfer_eth.rs [hunk 1]
```diff
@@ -1,47 +1,45 @@
...
-    println!(
-        "Transaction included in block {}",
-        receipt.block_number.expect("Failed to get block number")
-    );
+    // Send the transaction and listen for the transaction to be included.
+    let tx_hash = provider.send_transaction(tx).await?.watch().await?;
 
-    assert_eq!(receipt.from, alice);
-    assert_eq!(receipt.to, Some(bob));
+    println!("Send transaction... {}", tx_hash);
```
Comment (DaniPopes) [inline examples/transactions/examples/transfer_eth.rs:42]:
```suggestion
    println!("Sent transaction: {tx_hash}");
```

**Review by mattsse — APPROVED** [review]

File: examples/contracts/examples/deploy_from_artifact.rs [hunk 2]
```diff
@@ -50,9 +50,10 @@ async fn main() -> Result<()> {
     println!("Incremented number: {}", tx_hash);
 
     // Retrieve the number, which should be 43.
-    let Counter::numberReturn { _0 } = contract.number().call().await?;
+    let builder = contract.number();
+    let number = builder.call().await?._0;
 
-    println!("Retrieved number: {_0}");
+    println!("Retrieved number: {number}");
```
Comment (zerosnacks) [inline examples/contracts/examples/deploy_from_artifact.rs:56]: @DaniPopes there appears to be a bug here, I am unable to access `.number` like I can with the other examples (deploy_from_bytecode, deploy_from_contract)

Comment (zerosnacks) [reply examples/contracts/examples/deploy_from_artifact.rs:56]: I'll report it in `alloy-core` if I can narrow it down

**Comment by zerosnacks** [comment]
Merging as is, pending investigation: https://github.com/alloy-rs/examples/pull/98#discussion_r1634541752

**Pull request merged**